### PR TITLE
extended cloudify_resources plugin

### PR DESCRIPTION
**Pull request by jrzeszutek**
https://cloudifysource.atlassian.net/browse/CYBL-1570

### CODE REVIEW

**Comment by jrzeszutek** [comment]
I found out that it still needs some modifications... I'll reopen it soon

**Review by EarthmanT — COMMENTED** [review]
Please bump the version in CHANGELOG.txt and plugin.yaml.

**Comment by jrzeszutek** [comment]
> Please bump the version in CHANGELOG.txt and plugin.yaml.

Sure, bumped both to 1.25.7

File: cloudify_resources/tasks.py [hunk 1]
```diff
@@ -113,8 +135,124 @@ def reserve_list_item(ctx, **kwargs):
...
+            FREE_RESOURCES_LIST_PROPERTY)
+    reservations = \
+        ctx.instance.runtime_properties.get(
+            RESERVATIONS_PROPERTY, {})
+
+    if not reservation_id:
+        reservation_id = id_generator(size=8)
+        if reservation_id in reservations:
+            raise RecoverableError('reservation_id already taken.')
+    else:
```
Comment (ahmadiesa-abu) [inline cloudify_resources/tasks.py:214]: @jrzeszutek no need for the else just make the above code outside the if

**Review by ahmadiesa-abu — COMMENTED** [review]
small comment

File: cloudify_resources/tasks.py (hunk 1 above)
Comment (jrzeszutek) [reply cloudify_resources/tasks.py]: Done, thx @ahmadiesa-abu

**Review by ahmadiesa-abu — APPROVED** [review]

**Pull request merged**